### PR TITLE
[ODH-2262]: Posterserver logparsing added CRID id field

### PR DESCRIPTION
**Pull request by nilan3**
Updated log parser so crid id can be extracted from message field.

### CODE REVIEW

File: src/applications/log_parsing/poster_server/crid_event_creator.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import re
```
Comment (vunder-kind) [inline src/applications/log_parsing/poster_server/crid_event_creator.py]: This event_creator is unnecessary. Please use usual EventCreator class in driver.

File: src/applications/log_parsing/poster_server/driver.py [hunk 2]
```diff
@@ -30,13 +31,25 @@ def create_event_creators(configuration=None):
...
+            StringField("crid")
+        ]),
+        RegexpParser(r".*(?P<crid>crid[^\\]*)",
+                     return_empty_dict=True),
+        field_to_parse="message")
+
     return MatchField("source", {
         "PosterServer.Error.log": SourceConfiguration(
-            poster_server_log,
+            CompositeEventCreator()
```
Comment (needmorec) [inline src/applications/log_parsing/poster_server/driver.py]: You can create deduplicate this event creators and create a variable for it

File: src/applications/log_parsing/poster_server/driver.py [hunk 3]
```diff
@@ -30,13 +31,25 @@ def create_event_creators(configuration=None):
             r"^(?P<timestamp>\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}\,\d{3})"
             r"\s+(?P<level>\w+?)\s+(?P<module>\w+)\s+(?P<message>.*)"))
 
+    crid_creator = EventCreator(
+        Metadata([
+            StringField("crid")
+        ]),
+        RegexpParser(r".*(?P<crid>crid[^\\]*)",
```
Comment (vunder-kind) [inline src/applications/log_parsing/poster_server/driver.py]: Do we have crid in any message? Add substring matching option to EventCreator to filter all  messages without  word "crid"

Comment (nilan3) [reply src/applications/log_parsing/poster_server/driver.py]: @vunder-kind done

**Pull request merged**